feat: Add more context on data storage locations and API endpoints (#11652)
getsentry/sentry-docs: docs/api/index.mdx

- [Requests](/api/requests/)
- [Tutorial: Create a Sentry Authentication Token](/api/guides/create-auth-token/)
- [Tutorial: Create and List Teams with the Sentry API](/api/guides/teams-tutorial/)

> ## Finding the API base domain

Reviewer: ```suggestion
## Choosing the Right API Base Domain
```

> 
> While many of the API examples use `sentry.io` as the host for API endpoints, if
> you want to take advantage of [data storage location](/organization/data-storage-location/#what-types-of-data-are-stored-where) you should use the region
> specific domains instead.
> 
> - US region is hosted on `us.sentry.io`
> - DE region is hosted on `de.sentry.io`.
> 
> To find out which API resources are available on region base domains see [what
> types of data are stored
> where](/organization/data-storage-location/#what-types-of-data-are-stored-where) for more information.
> 

Reviewer: ```suggestion

While many of our APIs use `sentry.io` as the host for API endpoints, if
you want to indicate a specific [data storage location](/organization/data-storage-location/#what-types-of-data-are-stored-where), you should use region-specific domains.

- US region is hosted on `us.sentry.io`
- DE region is hosted on `de.sentry.io`.

To find out which API resources are available on region-based domains, see [what
types of data are stored
where](/organization/data-storage-location/#what-types-of-data-are-stored-where) for more information.

```
